enginetest/queries/create_table_queries.go

```diff
@@ -725,6 +725,138 @@ var CreateTableScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "create table with select preserves default",
+		SetUpScript: []string{
+			"create table a (i int primary key, j int default 100);",
+			"create table b (x int primary key, y int default 200);",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "create table t1 select * from a;",
+				Expected: []sql.Row{
+					{types.NewOkResult(0)},
+				},
+			},
+			{
+				Query: "show create table t1;",
+				Expected: []sql.Row{
+					{"t1", "CREATE TABLE `t1` (\n" +
+						"  `i` int NOT NULL,\n" +
+						"  `j` int DEFAULT '100'\n" +
+						") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin"},
+				},
+			},
+
+			{
+				Query: "create table t2 select j from a;",
+				Expected: []sql.Row{
+					{types.NewOkResult(0)},
+				},
+			},
+			{
+				Query: "show create table t2;",
+				Expected: []sql.Row{
+					{"t2", "CREATE TABLE `t2` (\n" +
+						"  `j` int DEFAULT '100'\n" +
+						") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin"},
+				},
+			},
+
+			{
+				Query: "create table t3 select j as k from a;",
```

Comment:
We sould maybe test `select j as i` also, to be safe. The other weird ones will be json_table and union/window/group by, but I'm not sure we really need those working, just not panicking.